src/provideInlineCompletionItems.ts

```diff
@@ -0,0 +1,71 @@
+import * as vscode from "vscode";
+import { AutocompleteResult, ResultEntry } from "./binary/requests/requests";
+import { completionIsAllowed } from "./provideCompletionItems";
+import runCompletion from "./runCompletion";
+import { COMPLETION_IMPORTS } from "./selectionHandler";
+
+const INLINE_REQUEST_TIMEOUT = 3000;
+
+export default async function provideInlineCompletionItems(
+  document: vscode.TextDocument,
+  position: vscode.Position
+): Promise<vscode.InlineCompletionList> {
+  try {
+    if (!completionIsAllowed(document, position)) {
+      return new vscode.InlineCompletionList([]);
+    }
+
+    const isEmptyLine = document.lineAt(position.line).text.trim().length === 0;
+
+    const response = await runCompletion(
```

Comment:
I had a code here that sends the request twice on empty line - where did it go

Reply:
this code was taken from [here](https://github.com/codota/tabnine-vscode/pull/706/files#diff-88c55a307a2aafd4bd44fa02d2ceb1a8fffa4246cde36e7cf4981e2df04b8e75R9) , where is the code u r referring to ?